webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/db/DBCommon.scala

```diff
@@ -125,8 +127,13 @@ trait DBCommon extends Specification with Loggable with BeforeAfterAll {
     }
   }
 
-  def cleanDb(): Unit = {
-    if (sqlClean.trim.size > 0) doobie.transactRunEither(xa => Update0(sqlClean, None).run.transact(xa)) match {
+  final def cleanDb(): Unit = {
+    doobie.transactRunEither(xa => {
+      val finalizer = fr"DROP SCHEMA temp_schema CASCADE".update.run.transact(xa)
+      ZIO.when(sqlClean.trim.size > 0)(
+        Update0(sqlClean, None).run.transact(xa)
+      ) *> finalizer
+    }) match {
```

Comment:
From what I understand this should not really be overriden, because it may contain strict finalizers in the supertrait. And the `sqlClean` was added to make it possible to still execute sql in inheritors.  
This should be the same for `initDb` but it has a more complex override in [TestDbMigration_5_6](https://github.com/Normation/rudder/blob/b5ce7c2f3f645b77512b60ec9db7f8f391d163bf/webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/migration/TestDbMigration.scala#L125), which needs to be rewritten.